### PR TITLE
`struct Rav1dFrameContext_task_thread`: Replace atomic intrinsics

### PR DESCRIPTION
**Pull request by rinon**

### CODE REVIEW

File: src/internal.rs [hunk 1]
```diff
@@ -430,12 +430,12 @@ pub(crate) struct Rav1dFrameContext_task_thread {
...
     pub num_tile_tasks: c_int,
-    pub init_done: atomic_int,
-    pub done: [atomic_int; 2],
+    pub init_done: AtomicI32,
+    pub done: [AtomicI32; 2],
     pub retval: Rav1dResult,
     pub update_set: bool, // whether we need to update CDF reference
-    pub error: atomic_int,
-    pub task_counter: atomic_int,
+    pub error: AtomicI32,
```
Comment (kkysen) [inline src/internal.rs:437]: I think we want this to be an `Atomic<Rav1dError>`.  We can `#[derive(Atom)]` on `enum Rav1dError` (using `atomig`).

Comment (kkysen) [reply src/internal.rs:437]: It might be fine to just do that later, though.

File: src/internal.rs [hunk 2]
```diff
@@ -516,7 +516,7 @@ pub(crate) struct Rav1dFrameContext {
     pub lf: Rav1dFrameContext_lf,
     pub task_thread: Rav1dFrameContext_task_thread,
     pub tile_thread: FrameTileThreadData,
-}
+    }
```
Comment (kkysen) [inline src/internal.rs:519]: Typo?

**Review by kkysen — APPROVED** [review]
LGTM besides that indentation typo.

File: src/internal.rs (hunk 1 above)
Comment (rinon) [reply src/internal.rs:436]: Definitely at some point, but that isn't the priority now or in the near future.

**Comment by rinon** [comment]
Added another atomic_i32 -> AtomicI32 commit, so worth a quick additional look.

**Review by kkysen — APPROVED** [review]
Still looks good with the `pending_tasks_merge` change now, but what was the hoist from `Rav1dFrameContext_task_thread_pending_tasks` to `Rav1dFrameContext_task_thread` for, (for some mutability reasons?)?  Can you add that to the PR description or commit message?

**Comment by rinon** [comment]
Updated the commit message. I moved it out in preparation for making the `pending_tasks` protected by a Mutex. The merge field was accessed outside the critical section (and it's atomic, so that's fine), so I needed to move it out of the struct that I planned to put into a Mutex.

**Pull request merged**